x-pack/plugins/security_solution/public/detections/components/callouts/missing_privileges_callout/translations.tsx

```diff
@@ -0,0 +1,147 @@
+import { MissingPrivileges } from './use_missing_privileges';
+
+export const MISSING_PRIVILEGES_CALLOUT_TITLE = i18n.translate(
+  'xpack.securitySolution.detectionEngine.missingPrivilegesCallOut.messageTitle',
+  {
+    defaultMessage: 'Insufficient privileges',
+  }
+);
+
+const CANNOT_EDIT_RULES = i18n.translate(
+  'xpack.securitySolution.detectionEngine.missingPrivilegesCallOut.cannotEditRules',
+  {
+    defaultMessage: 'Without that privilege you cannot create/edit detection engine rules.',
+  }
+);
+
+const CANNOT_EDIT_LISTS = i18n.translate(
+  'xpack.securitySolution.detectionEngine.missingPrivilegesCallOut.cannotEditLists',
+  {
+    defaultMessage: 'Without that privilege you cannot create/edit value lists.',
+  }
+);
+
+const CANNOT_EDIT_ALERTS = i18n.translate(
+  'xpack.securitySolution.detectionEngine.missingPrivilegesCallOut.cannotEditAlerts',
+  {
+    defaultMessage: 'Without that privilege you cannot change alert states (open or close alerts).',
+  }
+);
+
+export const missingPrivilegesCallOutBody = ({
+  indexPrivileges,
+  featurePrivileges,
+}: MissingPrivileges) => (
+  <FormattedMessage
+    id="xpack.securitySolution.detectionEngine.missingPrivilegesCallOut.messageBody.messageDetail"
+    defaultMessage="{essence} Missing privileges: {privileges} Related documentation: {docs}"
+    values={{
+      essence: (
+        <p>
+          <FormattedMessage
+            id="xpack.securitySolution.detectionEngine.missingPrivilegesCallOut.messageBody.essenceDescription"
+            defaultMessage="You are currently don't have all the required permissions. Some of the page functionality could be limited or unavailable. Please contact your administrator for further assistance."
+          />
+        </p>
+      ),
+      privileges: (
+        <ul>
+          {indexPrivileges.map((privilege, i) => (
+            <li key={i}>{missingIndexPrivileges(...privilege)}</li>
+          ))}
+          {featurePrivileges.map((privilege, i) => (
+            <li key={i}>{missingFeaturePrivileges(...privilege)}</li>
+          ))}
+        </ul>
+      ),
+      docs: (
+        <ul>
+          <li>
+            <DetectionsRequirementsLink />
+          </li>
+          <li>
+            <SecuritySolutionRequirementsLink />
+          </li>
+        </ul>
+      ),
+    }}
+  />
+);
+
+interface PrivilegeExplanations {
+  [key: string]: {
+    [privilegeName: string]: string;
+  };
+}
+
+const PRIVILEGE_EXPLANATIONS: PrivilegeExplanations = {
+  [SAVED_OBJECTS_MANAGEMENT_FEATURE_ID]: {
+    All: CANNOT_EDIT_RULES,
+  },
+  [DEFAULT_SIGNALS_INDEX]: {
+    write: CANNOT_EDIT_ALERTS,
+  },
+  [DEFAULT_LISTS_INDEX]: {
+    write: CANNOT_EDIT_LISTS,
+  },
+  [DEFAULT_ITEMS_INDEX]: {
+    write: CANNOT_EDIT_LISTS,
+  },
+};
+
+const getPrivilegesExplanation = (missingPrivileges: string[], index: string) => {
+  const explanationsByPrivilege = Object.entries(PRIVILEGE_EXPLANATIONS).find(([key]) =>
```

Comment:
Yea, it would work, but there is a subtle difference in types between the two. Consider the following examples:
```
const explanationsByPrivilege = Object.entries(PRIVILEGE_EXPLANATIONS).find(([key]) =>
  index.startsWith(key)
)?.[1];

const foo: string = explanationsByPrivilege?.foo;
```
It gives us a type error: `Type 'string | undefined' is not assignable to type 'string'.`
  
```
const explanationsByPrivilege =
  Object.entries(PRIVILEGE_EXPLANATIONS).find(([key]) => index.startsWith(key))?.[1] || {};

const foo: string = explanationsByPrivilege.foo;
```
In this case, TypeScript doesn't show any errors. However, `foo` could definitely be `undefined`, and we could get a runtime error down the road.
That's why I prefer using `?.`, as it is more explicit about possible `undefined` values.